internal/decryption/keyproviders/kmip.go

```diff
@@ -48,20 +49,46 @@ type KMIPEncryptedKey struct {
 
 // DecryptKey decrypts LEK using KMIP get or decrypt methods.
 func (ki *KMIPKeyIdentifier) DecryptKey(encryptedKey []byte) ([]byte, error) {
+	if len(ki.ServerNames) == 0 {
+		return nil, errors.New("server name is not provided")
+	}
+
 	kmipEncryptedKey, err := ki.decodeEncryptedKey(encryptedKey)
 	if err != nil {
 		return nil, err
 	}
 
-	kmipClient, err := ki.kmipClient()
-	if err != nil {
-		return nil, err
+	var clientError error
+	collectErr := func(err error, serverName string) {
+		if clientError == nil {
```

Comment:
[q] in which scenario, clientError would be different than nil? where are you setting this value?

Reply:
in the scenario where we have multiple KMIP servers, if the first one is failing then clientError will be initialized at this line. For the next servers that are failing will, the error will be concatenated with the previous clientError (that's why the name of the function, collectErr).